tests/filecheck/checks/check-line.test

```diff
@@ -0,0 +1,15 @@
+// RUN: sed -E "s/^(\/\/|;|#).*/ /" %s | grep -n foo | filecheck %s
+// using special command to remove comments instead of strip-comments.sh that
+// replaces comments with a space character, because we need to maintain the
+// line numbers in this special case
+test
+
+// CHECK: [[# @line + 1]]:foo
+foo
+
+foo
+// CHECK: [[# @line - 1]]:foo
+
+foo1
+foo2
+// CHECK: [[# @line - 2]]:foo1
```

Comment:
supernit: You are not testing the "easy" syntax of just `[[# @LINE]]`, I think we could add this to the test:

```
foo CHECK: [[# @line]]:foo
```

I'm also wondering if it would make sense to add a test along the lines of this:

```
// CHECK-NOT: [[# @LINE]]:foo
```

(because right now all tests would pass if the line number is substituted with nothing)